docs/docs-beta/docs/dagster-plus/deployment/branch-deployments/setting-up-branch-deployments.md

```diff
@@ -0,0 +1,484 @@
+- `AWS_REGION` - The AWS region where your ECR registry is located
+
+</TabItem>
+
+<TabItem value="gcr" label="Google Container Registry (GCR)">
+
+- `DAGSTER_CLOUD_API_TOKEN` - The Dagster+ agent token you created in [Step 2](#step-2-generate-a-dagster-agent-token)
+- `DAGSTER_CLOUD_URL` - Your Dagster+ base URL (`https://my_org.dagster.cloud`)
+- `GCR_JSON_KEY` - Your GCR JSON credentials
+
+</TabItem>
+
+</Tabs>
+
+### Step 4.4: Configure Gitlab CI/CD script
+
+:::note
+If you're using a Serverless deployment, this step isn't required. [Skip ahead](#step-45-verify-gitlab-pipeline-runs)
+:::
+
+In this step, you'll update the Gitlab CI/CD config to set up Docker registry access.
+
+In the `.gitlab-ci.yml` file, un-comment the `step` associated with your registry. For example, for the Gitlab container registry, you'd un-comment the following portion of the `.gitlab-ci.yml` file:
+
+```yaml
+build-image:
+  ...
+  before_script:
+    # For Gitlab Container Registry
+    - echo $CI_JOB_TOKEN | docker login --username $CI_REGISTRY_USER --password-stdin $REGISTRY_URL
+```
+
+Save and commit the files to the project.
+
+### Step 4.5: Verify Gitlab pipeline runs
+
+The last step is to verify that the Gitlab pipeline runs successfully.
+
+1. On the project page, click the **CI/CD** tab.
+2. In the list of pipelines, locate the latest branch deployment run. For example:
+
+![Show this in the UI](/img/placeholder.svg)
+
+  </TabItem>
+  <TabItem value="cli" label="dagster-cloud CLI">
+
+When the state of a branch or merge request is updated, Dagster+ first expects these steps to occur:
+
+Whenever the state of your branch is updated, Dagster+ expects the following steps to occur:
+
+1. A new image containing your code and requirements is built on the branch. Refer to [Managing code locations](/todo) to learn more.
+
+2. The new image is pushed to a Docker registry accessible to your agent.
+
+The details of how this is accomplished depend on your specific CI/CD solution.
+
+:::note
+
+The following examples assume the registry URL and image tag are stored in the `LOCATION_REGISTRY_URL` and `IMAGE_TAG` environment variables.
+
+:::
+
+### Step 4.1 Create a branch deployment associated with the branch
+
+Execute the following command within your CI/CD process:
+
+   ```shell
+   BRANCH_DEPLOYMENT_NAME=$(
+       dagster-cloud branch-deployment create-or-update \
+           --organization $ORGANIZATION_NAME \
+           --api-token $DAGSTER_CLOUD_API_TOKEN \ # Agent token from Step 1
+           --git-repo-name $REPOSITORY_NAME \ # Git repository name
+           --branch-name $BRANCH_NAME \ # Git branch name
+           --commit-hash $COMMIT_SHA \ # Latest commit SHA on the branch
+           --timestamp $TIMESTAMP # UTC unixtime timestamp of the latest commit
+   )
+   ```
+
+One or more additional parameters can optionally be supplied to the `create-or-update` command to enhance the Branch Deployments UI in Dagster+:
```

Comment:
maybe this would be better formatted as a table with the arg and the description, rather than inline comments? would defer to docs folks on their opinions, but just a thought i had

Reply:
Will also defer to others with opinions -- seems reasonable, but there's something nice about being able to just copy and paste the command

Reply:
i didn't consider copy-paste ability. that's a strong reason to keep as is